lib/src/chart/radar_chart/radar_chart_helper.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:equatable/equatable.dart';
+import 'package:fl_chart/fl_chart.dart';
+import 'package:fl_chart/src/utils/list_wrapper.dart';
+
+/// Contains all the function to help the RadarChart work
+class RadarChartHelper {
+  /// Contains List of cached results, based on [List<RadarDataSet>]
+  ///
+  /// We use it to prevent redundant calculations
+  static final Map<ListWrapper<RadarDataSet>, RadarChartMinMaxAxisValues>
+      _cachedResults = {};
+
+  static RadarChartMinMaxAxisValues calculateMinMaxAxisValue(
```

Comment:
Why don't you just compute this in the `RadarChartData` contructor ?
with something like this : 
```dart
min = min ?? dataSets?.expand((s) => s.dataEntries).max.value ?? 0,
max = max ?? dataSets?.expand((s) => s.dataEntries).min.value ?? 0,
```
with this extension method or inline if you prefer:
```dart
extension DataSetLimits on Iterable<RadarEntry> {
  RadarEntry get max => reduce((a, b) => a.value > b.value ? a : b);
  RadarEntry get min => reduce((a, b) => a.value < b.value ? a : b);
}
```
I know that it generate one more iteration than your solution but it's a bit less complex and that's better than the previous implementation using a custom getter reevaluated generating 3 iterations per rendering

Reply:
Check my answer [here](https://github.com/imaNNeoFighT/fl_chart/pull/995#discussion_r941729184)